dask/dataframe/io/tests/test_parquet.py

```diff
@@ -3163,6 +3163,30 @@ def test_partitioned_no_pandas_metadata(tmpdir, engine, write_cols):
     assert_eq(result[list(expect.columns)], expect, check_index=False)
 
 
+@PYARROW_MARK
+def test_pyarrow_directory_partitioning(tmpdir):
```

Comment:
@jcrist - I mentioned a few weeks back that thee pyarrow.dataset API is capable of supporting more partitioning options than we are currently exposing in `read_parquet`. This PR mostly resolves this.